lambda/docker-runner.yaml

```diff
@@ -0,0 +1,202 @@
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 09 ? * MON *)  # Run every Monday at 9am
+    ConstraintDescription: "Use schedule format: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
+  SynapsePAT:
+    Type: String
+    NoEcho: true
+    Default: ""
+    Description: >
+      (Optional) The Synapse personal access token (PAT).
+      https://docs.synapse.org/rest/POST/personalAccessToken.html
+Conditions:
+  HasEntryPoints: !Not
+    - !Equals
+      - !Join [ "", !Ref EntryPoints ]
+      - ''
+  HasCommand: !Not
+    - !Equals
+      - !Join ["", !Ref Command]
+      - ''
+  HasWorkingDirectory: !Not [!Equals [!Ref WorkingDirectory, ""]]
+  HasSynapsePAT: !Not [!Equals [!Ref SynapsePAT, ""]]
+Resources:
+  FileSystemResource:
+    Type: 'AWS::EFS::FileSystem'
+    Properties:
+      PerformanceMode: !Ref PerformanceMode
+      Encrypted: !Ref Encrypted
+      FileSystemPolicy:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action:
+              - "elasticfilesystem:ClientMount"
+            Principal:
+              AWS: "*"
+  MountTarget:
+    Type: AWS::EFS::MountTarget
+    Properties:
+      FileSystemId: !Ref FileSystemResource
+      SubnetId: !ImportValue {'Fn::Sub': '${AWS::Region}-docker-runner-vpc-MountTargetSubnet'}
+      SecurityGroups: [!ImportValue {'Fn::Sub': '${AWS::Region}-docker-runner-vpc-DefaultSecurityGroup'}]
+  AccessPoint:
+    DependsOn: [MountTarget]
+    Type: 'AWS::EFS::AccessPoint'
+    Properties:
+      FileSystemId: !Ref FileSystemResource
+      PosixUser:
+        Uid: "1000"
+        Gid: "1000"
+      RootDirectory:
+        CreationInfo:
+          OwnerGid: "1000"
+          OwnerUid: "1000"
+          Permissions: "0777"
+        Path: !Ref EfsPath
+  EfsPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AmazonElasticFileSystemClientFullAccess
+            Effect: Allow
+            Action:
+              - elasticfilesystem:ClientMount
+              - elasticfilesystem:ClientRootAccess
+              - elasticfilesystem:ClientWrite
+              - elasticfilesystem:DescribeMountTargets
+            Resource: !GetAtt FileSystemResource.Arn
+  DockerRunnerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Timeout: 900
+      PackageType: Image
+      ImageUri: !Ref ImageUri
+      ImageConfig:
+        Command: !If [HasCommand, !Ref Command, !Ref 'AWS::NoValue']
+        EntryPoint: !If [HasEntryPoints, !Ref EntryPoints, !Ref 'AWS::NoValue']
+        WorkingDirectory: !If [HasWorkingDirectory, !Ref WorkingDirectory, !Ref 'AWS::NoValue']
+      MemorySize: !Ref MemorySize
+      Environment:
+        Variables:
+          EFS_PATH: !Ref EfsPath
+          SYNAPSE_PAT: !If [HasSynapsePAT, !Ref SynapsePAT, !Ref 'AWS::NoValue']
```

Comment:
The Synapse secret (each input env variable) is encrypted with KMS using the default key for this account, sent to wherever the lambda runs, and decrypted by the lambda runner when invoked.

Reply:
there is no KMS encryption going on.

Reply:
AWS uses KMS when deploying the lamda